app/src/main/java/com/example/pokedexapp/ui/pokemon_detail_screen/BaseStatsGraph.kt

```diff
@@ -0,0 +1,99 @@
+package com.example.pokedexapp.ui.pokemon_detail_screen
+
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.layout.Layout
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+
+
+@Composable
+fun BaseStatsGraph(
+    spaceBetweenVertically: Dp = 0.dp,
+    spaceBetweenHorizontally: Dp = 0.dp,
+    baseStatsIndex: Int,
+    baseStatLabel: @Composable (index: Int) -> Unit,
+    baseStatBar: @Composable (index: Int) -> Unit,
+    baseStatValue: @Composable (index: Int) -> Unit,
+    modifier: Modifier = Modifier
+
+) {
+    val baseStatLabel = @Composable { repeat(baseStatsIndex) { baseStatLabel(it) } }
+    val baseStatBar = @Composable { repeat(baseStatsIndex) { baseStatBar(it) } }
+    val baseStatValue = @Composable { repeat(baseStatsIndex) { baseStatValue(it) } }
+
+    Layout(
```

Comment:
O que te motivou a usar essa abordagem pra criar o componente ? Olhando tua UI depois de buildar o app parecia ser mais fácil só criar uma célula pra ser repetida em cada linha dentro de uma Collumn.

Reply:
Primeiro acho que quis me desafiar um pouco pra tentar entender como funcionam esses custom layouts, e achei bem bacana. Entendo que por questão de legibilidade e manutenção não é muito bom utilizar essa abordagem, só se for a única alternativa. Mas, além disso, a motivação foi fazer o alinhamento vertical dos textos com a barra. Por exemplo, se aumentar a altura da barra, os textos vão ficar centralizados nela.

Reply:
Entendi, nesse projeto faz sentido tu testar mais coisas diferentes e se aventurar mais. Mas cuida quando estiver alocado em projeto de clientes por que vai ser melhor se for possível usar algo mais simples pelas mesmas questões qu tu acabou de citar mas também pelos prazos supondo que tu levou algum tempo extra pra fazer esse esquema todo. Tem um termo que é sempre bom ter em mente que é o over-engineering, basicamente uma referência a quando a gente pode fazer algo mais simplificado mas faz uma engenharia extra. É algo pra se evitar. Acho que fica de feedback pro uso da selead class naquele outro contexto em que comentamos antes, me parecia que daria pra simplificar também. De qualquer forma vou deixar approvada a PR e numa próxima implementação é só ter em mente essas questões.